### PR TITLE
Ignore untracked files with ignore_unstaged_changes parameter

### LINKED ISSUE #1075: Allow to also ignore untracked files
| Q               | A
| --------------- | ---
| Version         | 1.15.0
| Bug?            | yes/no
| New feature?    | no
| Question?       | no
| Documentation?  | no
| Related tickets | N/A

I faced an issue for which ignoring unstaged changes was not sufficient: running a unit test suite as precommit with an untracked unit test file. it wasn't expected to commit that file at that moment, but it existed already.
And so being on the file system it was taken into account by PHPUnit.

I was about to quickly contribute a new `ignore_untracked_files` option, but I realized that it may not be good to have it as a separate configuration, as it would mean that it could be used alone, without ignore_unstaged_changes.

Another solution is to simply add `--ignore-untracked` to the stach command, but maybe this is a BC break?

So I'm questioning what would be the best way to solve the issue I faced.

Note: issue faced when doing lot of code changes then trying to do clean atomic commits 🙂

### PR DESCRIPTION
**Pull request by yguedidi**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any

closes #1075

### CODE REVIEW

File: src/Event/Subscriber/StashUnstagedChangesSubscriber.php [hunk 1]
```diff
@@ -97,14 +97,32 @@ public function handleErrors(): void
...
+        $hasUntracked = \count($this->repository->getWorkingCopy()->getUntrackedFiles()) > 0;
+
+        if (!$hasPending && !$hasUntracked) {
             return;
         }
 
         try {
-            $this->io->write(['<fg=yellow>Detected unstaged changes... Stashing them!</fg=yellow>']);
-            $this->repository->run('stash', ['save', '--keep-index', uniqid('grumphp')]);
+            if ($hasPending && !$hasUntracked) {
```
Comment (veewee) [inline src/Event/Subscriber/StashUnstagedChangesSubscriber.php:108]: Does it make sense to add 3 different messages here?
Would 1 suffice as well?

**Comment by veewee** [comment]
Looks good, added a small question.

File: src/Event/Subscriber/StashUnstagedChangesSubscriber.php (hunk 1 above)
Comment (veewee) [reply src/Event/Subscriber/StashUnstagedChangesSubscriber.php:108]: Ok, lets keep it like this then!

**Comment by veewee** [comment]
Thanks!

**Comment by yguedidi** [comment]
You're welcome! Thanks for the quick merge

**Pull request merged**